meet_balls_app/tests.py

```diff
@@ -64,6 +76,101 @@ def test_successful_logout(self, client, player):
+            'password': 'testpassword',
+            'confirm_password': 'testpassword',
+            'birthdate': '1990-01-01',
+            'favorite_ball_game': 'Soccer'
+        }
+
+        register_url = reverse('registerUser')
+        response = client.post(register_url, data)
+        assert Player.objects.filter(user__username='testuser').exists()
+        assert Player.objects.filter(user__username='testuser').count() == 1
+        assert response.status_code == 302
+        assert response.url == reverse('loginUser')
+
+    @pytest.mark.parametrize('username, password, confirm_password, birthdate, favorite_ball_game', [
+        ('', 'passwrod', 'password', '1990-01-01', 'Soccer'),
+        ('testuser', '', 'password', '1990-01-01', 'Soccer'),
+        ('testuser', 'passwrod', '', '1990-01-01', 'Soccer'),
+        ('testuser', 'passwrod', 'password', '', 'Soccer'),
+        ('testuser', 'passwrod', 'password', '1990-01-01', '')
+    ])
+    def test_unsuccessful_register_blank_field(
+            self, client, username, password, confirm_password, birthdate, favorite_ball_game):
+        data = {
+            'username': username,
+            'password': password,
+            'confirm_password': confirm_password,
+            'birthdate': birthdate,
+            'favorite_ball_game': favorite_ball_game
+        }
+        response = client.post('/register/', data)
+
+        assert response.status_code == 200
+        assert 'All fields are required.' in response.content.decode()
+        assert response.templates[0].name == 'meet_balls_app/register.html'
+
+    def test_unsuccessful_register_user_is_already_exist(self, client):
+        data = {
+            'username': 'user1',
+            'password': 'password',
+            'confirm_password': 'password',
+            'birthdate': '1990-01-01',
+            'favorite_ball_game': 'Soccer'
+        }
+        response = client.post('/register/', data)
+
+        assert response.status_code == 200
+        assert 'Username already exists.' in response.content.decode()
+        assert response.templates[0].name == 'meet_balls_app/register.html'
+
+    def test_unsuccessful_register_passwords_dont_match(self, client):
+        data = {
+            'username': 'testuser',
+            'password': 'password',
+            'confirm_password': 'password1',
+            'birthdate': '1990-01-01',
+            'favorite_ball_game': 'Soccer',
+        }
+
+        response = client.post('/register/', data)
+
+        assert response.status_code == 200
+        assert 'Passwords do not match.' in response.content.decode()
+        assert response.templates[0].name == 'meet_balls_app/register.html'
+
+    def test_unsuccessful_register_birthdate_in_future(self, client):
+        future_date = timezone.now() + datetime.timedelta(days=1)
+        data = {
+            'username': 'testuser',
+            'password': 'password',
+            'confirm_password': 'password1',
```

Comment:
I think the `'confirm_password': 'password1,'` should be here `'confirm_password': 'password',`

Reply:
Agree. fixed.